### DISCUSSION TITLE
Investigating and choosing a group message software application

### DISCUSSION THREAD

**Discussion by ai-ml-robots** [ai-ml-robots/ai-ml-robots.github.io#1] [General]
## 👋 Welcome !

At last night's meeting (2024-08-21), determining the best way to have conversations in-between the regular monthly meetings ... was identified as the most important goal to immediately improve the group.

My personal opinion is to minimise the number of different software applications and tools we have to learn and maintain ... to be part of **AI + ML + Robots**.  Also preferable to have good integration between all our needs and the technical solutions.  _Can we use GitHub for as much as possible ... to run our group ?_

Whilst Slack and Discord are very common choices ... both have significant disadvantages ...
* Slack (free-tier) hides messages after 90 days
* Discord has a pretty poor user experience

As an experiment, I'd like to try GitHub Discussions for open-ended conversation ... and [GitHub Issues](https://github.com/ai-ml-robots/ai-ml-robots.github.io/issues) for specific group actions ...
* [GitHub Discussions feature overview](https://github.com/features/discussions)
* [GitHub Discussions documentation](https://docs.github.com/en/discussions)

What do you think ?

**Comment by geekscape** [comment]
Bugzilla is a blast-from-the-past ... unfortunately, it had a poor user interface.
Fortunately, these days issue trackers are getting better ... and with GitHub and GitLab now have tight integration with the rest of the project development process.
The main challenge will be to curate the discussions (a little) and manage the issues (more so) ... such that we don't drown under too many.  Best to keep them under control and ensure that good communication and progress is being made.

**Comment by kwdChan** [comment]
Hi Andy, thanks again for organising the meeting space! GitHub Discussion seems like a great choice for organised discussion to me but I am not sure if it's suitable for casual interactions - like saying stupid/funny things and sharing random ideas. I think it's still important to have some instant messaging methods if we want to have these kinds of casual interactions. 

But I think this is great for sharing project progress. Each project can be its own conversation in the show and tell channel and not get lost in the sea of messages in a messaging app. I will spend some time to write an intro for my project and share it soon. 

I'm Daniel btw

**Comment by geekscape** [comment]
hi Daniel, I agree ... GitHub Discussions are not designed to be used for casual interactions (messaging).  It is not light-weight, but also not too heavy-weight ... pretty much like an old-school Internet forum.

For sharing your project (details and progress), I've renamed the Discussion category "Show and tell" to "Project (show and tell)" ... a minor change, yet provides some emphasis on the "project" aspect :smile:
I'm really looking forward to seeing your robot project show-and-tell and on-going progress (and @fareed1983's too) !

<personal opinion/> I've found on-line community groups (Discord, for example) tend to drown under too many channels of re-posted Internet memes !  Because it can be too easy to post every random idea or link (GIF89 image, video) that pops into someone's head :rofl: 

In regards to instant messaging, **I've set-up a Signal group "AI + ML + Robots"** ... I'm thinking that will be useful for ad-hoc, time critical chatter and queries, e.g quick turn-around for idea discussion, local events or someone wanting to be let into the meeting venue (with a sign on the door with a QR code to the Signal group).  Let's see how that is used and evolves.

**Join Signal instant messaging group: AI + ML + Robots ...
https://signal.group/#CjQKIGXJXvszdtKjrNhLw517IGrDB3XHOHsRlNjAuqveI1UCEhC98bWmG_huGi9Vm2AE2OLK**

If something isn't working or we have a gap ... then, let's keep iterating !
